python/katana/CMakeLists.txt

```diff
@@ -37,54 +37,66 @@ function(generate_cython_source BASE)
   generate_cython_file(${BASE}.pyx DEPENDS ${_args_DEPENDS})
 endfunction()
 
-add_cython_target(galois galois.pyx CXX OUTPUT_VAR GALOIS_SOURCES)
-add_library(galois MODULE ${GALOIS_SOURCES})
-python_extension_module(galois)
-target_link_libraries(galois Katana::galois)
+set(KATANA_PYTHON_ROOT ${CMAKE_CURRENT_SOURCE_DIR})
 
-add_cython_target(numpy_structs numpy_structs.pyx CXX OUTPUT_VAR NUMPY_STRUCTS_SOURCES)
-add_library(numpy_structs MODULE ${NUMPY_STRUCTS_SOURCES})
-python_extension_module(numpy_structs)
+function(add_cython_module NAME)
+  set(options)
```

Comment:
perhaps `no_value_options` `single_value_options` `multi_value_options`?

Reply:
Agreed. `options`/`options1`/`optionsmany` appear to be used through out our code base (along with other odd variants). Should I change them all?

Reply:
It's up to you.

My main nit is `options1` seems too cute and `optionsmany` should at least have a `_` somewhere.

We should at least fix that aspect for this PR.

If you did make everything consistent, I'd be grateful, as I don't expect most developers are going to be `cmake`-gurus so making the cmake we do write more self-documenting is more important than usual.

OTOH, it is a battle against entropy....

Reply:
I'll do it for the open repo for the moment. I actually really like fixing things like this. I just try to avoid useless changes throughout the repo.